orderer/common/multichannel/registrar.go

```diff
@@ -1044,3 +1057,15 @@ func (r *Registrar) removeSystemChannel() error {
 
 	return nil
 }
+
+func (r *Registrar) removeLedgerAsync(channelID string) {
+	go func() {
+		r.pendingRemoval.Store(channelID, true)
```

Comment:
insert into the map[string]bool before launching the goroutine:
```
pendingRemoval[channelID] = true
go func() { ... //remove from ledger }
```

pendingRemoval is protected by the `r.lock` here.

Reply:
🧹